skipchain/skipchain.go

```diff
@@ -448,9 +448,7 @@ func (s *Service) OptimizeProof(req *OptimizeProofRequest) (*OptimizeProofReply,
 			err := sendForwardLinkRequest(sb.Roster, req, reply)
 
 			if err != nil {
-				log.Error("could not create a missing forward link:", err)
-				// reset the index to try to create lower levels
-				index = sb.Index
+				log.Warn("could not create missing forward link, trying next block:", err)
```

Comment:
Like I said in the PR, if you don't reset the index, you will miss potential forward-links because it will assume the forward-link has been created and the returned proof will be wrong.

Reply:
hmm, if I brain-trace the code, it will simply add the current block as is, so without the new forward-link, and then go on to the next block. But perhaps my brain-trace-capability is in Friday mode. I just checked it again, and I'm still sure it should work.

Reply:
```go
if sb.Index < index {
    // Skip blocks thanks to the new forward-link.
    continue
}
```

If the forward-link fails and index keep the target value, it will skip the following blocks of the previous proof until the target index. **But** the forward-link doesn't exist as it failed.

Reply:
Friday brain...